compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -1502,6 +1502,28 @@ LogicalResult initGPULaunchConfig(ModuleOp moduleOp) {
     auto exportOp = exportOps.lookup(funcOp.getName());
     if (!exportOp)
       continue;
+
+    // First check whether we already have workgroup count set--it's a
+    // "contract" to indicate that we should bypass all tiling and
+    // distribution to go down just the most basic lowering flow.
+    if (Block *body = exportOp.getWorkgroupCountBody()) {
+      auto retOp = cast<IREE::HAL::ReturnOp>(body->getTerminator());
+      // For scalar dispatch cases--using just one thread of one workgroup.
+      auto isOne = [](Value value) { return matchPattern(value, m_One()); };
+      if (llvm::all_of(retOp.getOperands(), isOne)) {
+        std::array<int64_t, 3> workgroupSize = {1, 1, 1};
+        if (failed(setDispatchConfig(funcOp, workgroupSize, std::nullopt)))
+          return failure();
+        auto translationInfo = IREE::Codegen::TranslationInfoAttr::get(
+            funcOp.getContext(),
+            IREE::Codegen::DispatchLoweringPassPipeline::LLVMGPUBaseLowering);
+        if (failed(setTranslationInfo(funcOp, translationInfo))) {
+          return failure();
+        }
+        continue;
+      }
+    }
```

Comment:
Can you add a test to `LLVMGPU/test/llvmgpu_pipeline_test.mlir` (or some place like that) showing what dispatch this is solving? Is this just for scalar dispatches?

Reply:
I was thinking the test would be basically the same as https://github.com/openxla/iree/pull/14377/files#diff-24b6180b258e5500ab5f5385e05b06095ac5fb6c28bf469369ec38d5b519c145, so was not sure if I should add it.

Reply:
Yes let's add a test like the above into `LLVMGPU/test/ROCDL` and use `iree-rocdl-lower-executable-target-pass`.